sycl/test/e2e_test_requirements/no-unsupported-without-info.cpp

```diff
@@ -0,0 +1,547 @@
+// This test is intended to ensure that we have no tests marked as
+// UNSUPPORTED without an information added to a test.
+// For more info see: sycl/test-e2e/README.md
+//
+// The format we check is:
+// UNSUPPORTED: lit,features
+// UNSUPPORTED-TRACKER: [GitHub issue URL|Internal tracker ID]
+// *OR*
+// UNSUPPORTED: lit,features
+// UNSUPPORTED-INTENDED: explanation why the test isn't intended to be run with this feature
+//
+// GitHub issue URL format:
+//     https://github.com/owner/repo/issues/12345
+//
+// Internal tracker ID format:
+//     PROJECT-123456
+//
+// REQUIRES: linux
+//
+// Explanation of the command:
+// - search for all "UNSUPPORTED" occurrences, display line with match and the next one
+//   -I, --include to drop binary files and other unrelated files
+// - in the result, search for "UNSUPPORTED" again, but invert the result - this
+//   allows us to get the line *after* UNSUPPORTED
+// - in those lines, check that UNSUPPORTED-TRACKER or UNSUPPORTED-INTENDED is
+//   present and correct. Once again, invert the search to get all "bad" lines
+//   and save the test names in the temp file
+// - make a final count of how many ill-formatted directives there are and
+//   verify that against the reference
+// - ...and check if the list of improperly UNSUPPORTED tests needs to be updated.
+//
+// RUN: grep -rI "UNSUPPORTED:" %S/../../test-e2e \
+// RUN: -A 1 --include=*.c --include=*.cpp --no-group-separator | \
```

Comment:
`.c` include is a certain legacy from `XFAIL` test: I'm not entirely sure if we ever should have any `.c` tests. SYCL is C++17

That's minor, we can address it in a follow-up PR for both `XFAIL` and `UNSUPPORTED`

Reply:
Looks like there are no `.c` tests in test-e2e at all.

Reply:
> Looks like there are no `.c` tests in test-e2e at all.

Yeah, there shouldn't be. I probably followed `suffixes` which we have listed in `lit.cfg.py`. In any case I don't think that it is possible to use `-fsycl` in C mode, or use SYCL headers in `.c` mode, so I think that we can do a separate cleanup everywhere to explicitly say that `.c` is not considered as a test for us and drop it from XFAIL/UNSUPPORTED checks